aws/container-linux/kubernetes/ami.tf

```diff
@@ -14,6 +23,27 @@ data "aws_ami" "coreos" {
 
   filter {
     name   = "name"
-    values = ["CoreOS-${var.os_channel}-*"]
+    values = ["CoreOS-${local.channel}-*"]
+  }
+}
+
+data "aws_ami" "flatcar" {
+  most_recent = true
+  owners      = ["075585003325"]
+
+  filter {
+    name   = "architecture"
+    values = ["x86_64"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  filter {
+    name   = "name"
+    values = ["Flatcar-${local.channel}-*"]
   }
 }
+
```

Comment:
I'm not familiar with typhoon & HCL, but am just curious.
Is there a good way to reduce number data definitions?
It seems that a large part of `ami.tf` is defined again in `workers/ami.tf`.

Reply:
Its definitely possible with an output and then consumption of that output.

The main question is whether we _should_. The worker inner module can be used independently for worker pools, so it can't refer to the outer module, so it must have an ami.tf contents. Having the outer kubernetes module use the worker module both to create workers (purpose of the module) _and_ to pick an AMI for controllers feels odd and kinda wrong. Right now, both modules pick an AMI for the instances they create and they happen to pick AMIs in the same way. Favoring some duplication over weird coupling.